create_builder.go

```diff
@@ -239,6 +239,9 @@ func (c *Client) addBuildpacksToBuilder(ctx context.Context, opts CreateBuilderO
 				return errors.Wrapf(err, "extracting from registry %s", style.Symbol(b.URI))
 			}
 		case buildpack.URILocator:
+			if b.URI == "" {
```

Comment:
nit: I see that we get `locator` from `b.URI` in line 203, then we mutate it based on the deprecated key. To make this more comprehensible couldn't we do one of the following: set `b.URI` to locator in line 206 OR use `locator` only after line 203.

Reply:
I wouldn't want to do the first recommendation, because we may only want to replace `b.URI` _if_ we know the locator is a `URILocator`. I opted not to use `locator` only after line 243 because it would break the encapsulation provided by the `b` object (which I believe is of `BuildpackConfig` type).

So there's tradeoffs all around, but i think this is best approach.